### PR TITLE
chore: improve loggings with debug and searched files

### PR DESCRIPTION
**Pull request by Shahroz16**
This PR is based on [discussion](https://github.com/customerio/cio-sdk-tools/pull/17#discussion_r1310000037) and early comments by TS. 

Since when we search for a regex, we do it in particular files that can serve the purpose, if there are different ways customers has initialized our SDK, they would get an error of "SDK not initialized". So this PR is facilitating that problem.

### CODE REVIEW

**Comment by levibostian** [comment]
Could you include a description explaining why this PR was made? What problem it's trying to solve? 

Knowing this would help a lot with reviewing this PR. Thanks!

File: src/checks/ios.ts [hunk 1]
```diff
@@ -332,10 +332,28 @@ async function validateSDKInitialization(project: iOSProject): Promise<void> {
...
+    logger.debug(`Search Criteria:`);
+    logger.debug(
+      `Target File Names: ${sdkInitializationFiles.formattedTargetFileNames}`
+    );
+    logger.debug(
+      `Target File Patterns: ${sdkInitializationFiles.formattedTargetPatterns}`
+    );
+    logger.debug(
+      `Looked into the following files: ${sdkInitializationFiles.formattedSearchedFiles}`
+    );
```
Comment (mrehan27) [inline src/checks/ios.ts:349]: I don't have more ideas at the moment, but would like to improve logs here. Something like this, may be?

```
Searched files with names ${sdkInitializationFiles.formattedTargetFileNames}
Searched files with keywords ${sdkInitializationFiles.formattedTargetPatterns}
```

**Review by mrehan27 — APPROVED** [review]

**Review by levibostian — COMMENTED** [review]
Looks good to me. I would prefer if Rehan approves though since he has more comments on the logic.

File: src/checks/react_native.ts [hunk 2]
```diff
@@ -69,18 +71,45 @@ async function validateSDKInitialization(
...
+        'index',
+        'init',
+        'setup',
+        'config',
+        'start',
+        'main',
+        'route',
+        'navigation',
+        'provider',
+      ],
```
[the remark this answers is not in the document]
Comment (Shahroz16) [reply src/checks/react_native.ts:85]: Even though `config.js` can be a name, i feel like they might make more sense in `targetFilePatterns`

File: src/utils/file.ts [hunk 3]
```diff
@@ -74,10 +74,16 @@ export function shouldIgnoreDirectory(
     'icons',
     'fonts',
   ];
+  // Directory names that contain these substrings should be ignored
+  const excludedSubstrings = ['temp', 'cache', 'test'];
```
[the remark this answers is not in the document]
Comment (Shahroz16) [reply src/utils/file.ts:78]: I get what you are saying, that's a real concern but I think these options are relatively safe as I don't think any src directory contains these strings in the path.

File: src/checks/ios.ts (hunk 1 above)
Comment (Shahroz16) [reply src/checks/ios.ts:349]: `Searching` might make more sense.

**Pull request merged**